pkg/operator/controllers/machine/machine_controller_test.go

```diff
@@ -0,0 +1,322 @@
+			arocli: newFakeAro(&baseCluster),
+			wantConditions: []status.Condition{{
+				Type:    arov1alpha1.MachineValid,
+				Status:  corev1.ConditionTrue,
+				Message: "All machines valid",
+				Reason:  "CheckDone",
+			}},
+		},
+		{
+			name:   "wrong_vm_size",
+			maocli: newFakeMao("512", "azureopenshift", "Standard_D4s_v9"),
+			arocli: newFakeAro(&baseCluster),
+			wantConditions: []status.Condition{{
+				Type:    arov1alpha1.MachineValid,
+				Status:  corev1.ConditionFalse,
+				Message: "machine foo-hx8z7-worker-0: invalid VM size 'Standard_D4s_v9'",
+				Reason:  "CheckFailed",
+			}},
+		},
+		{
+			name:   "wrong_disk_size",
+			maocli: newFakeMao("64", "azureopenshift", "Standard_D4s_v3"),
+			arocli: newFakeAro(&baseCluster),
+			wantConditions: []status.Condition{{
+				Type:    arov1alpha1.MachineValid,
+				Status:  corev1.ConditionFalse,
+				Message: "machine foo-hx8z7-worker-0: invalid disk size '64'",
+				Reason:  "CheckFailed",
+			}},
+		},
+		{
+			name:   "wrong_image_publisher",
+			maocli: newFakeMao("512", "bananas", "Standard_D4s_v3"),
+			arocli: newFakeAro(&baseCluster),
+			wantConditions: []status.Condition{{
+				Type:    arov1alpha1.MachineValid,
+				Status:  corev1.ConditionFalse,
+				Message: "machine foo-hx8z7-worker-0: invalid image '{bananas aro4   }'",
+				Reason:  "CheckFailed",
+			}},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r := &MachineReconciler{
+				maocli:                 tt.maocli,
+				log:                    logrus.NewEntry(logrus.StandardLogger()),
+				arocli:                 tt.arocli,
+				isLocalDevelopmentMode: false,
+				role:                   "master",
+			}
+
+			_, err := r.Reconcile(context.Background(), tt.request)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Machine.Reconcile() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+
+			cluster, err := r.arocli.AroV1alpha1().Clusters().Get(context.Background(), arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+			if err != nil {
+				t.Fatal("Error found")
+			}
+
+			if !cmp.Equal(cluster.Status.Conditions[0].Type, tt.wantConditions[0].Type) {
+				t.Errorf("Unexpected Type found\n want: %v\n got: %v", tt.wantConditions[0].Type, cluster.Status.Conditions[0].Type)
+			}
+
+			if !cmp.Equal(cluster.Status.Conditions[0].Status, tt.wantConditions[0].Status) {
+				t.Errorf("Unexpected Status found\n want: %v\n got: %v", tt.wantConditions[0].Status, cluster.Status.Conditions[0].Status)
+			}
+
+			if !cmp.Equal(strings.TrimSpace(cluster.Status.Conditions[0].Message), tt.wantConditions[0].Message) {
+				t.Errorf("Unexpected Message found\n want: %v\n got: %v", tt.wantConditions[0].Message, cluster.Status.Conditions[0].Message)
+			}
+
+			if !cmp.Equal(cluster.Status.Conditions[0].Reason, tt.wantConditions[0].Reason) {
+				t.Errorf("Unexpected Reason found\n want: %v\n got: %v", tt.wantConditions[0].Reason, cluster.Status.Conditions[0].Reason)
+			}
```

Comment:
@m1kola and I spoke about this, but for others, the reason that there are multiple test assertions in this test rather than one is because `cluster.Status.Conditions` (which is vendored in) outputs `LastTransitionTime`, so tests always fail when attempting to compare both arrays because there's always a difference in timestamp. There's no simple way to exclude this, so we are leaving it as is (with simplified assertions) for the time being.